sources/web/datalab/auth.ts

```diff
@@ -30,16 +31,35 @@ var oauth2Client: any = undefined;
 // These are the gcloud credentials and are not actually secret.
 let clientId = '32555940559.apps.googleusercontent.com';
 let clientSecret = 'ZmssLNjJy2998hD4CTg2ejr2';
-let gcloudDir = '/content/datalab/.config';
-let userCredFile = gcloudDir + '/credentials';
-let appCredFile = gcloudDir + '/application_default_credentials.json';
-let botoFile = '/etc/boto.cfg';
+
+// The application settings instance.
+var appSettings: common.Settings;
 
 let scopes = [
   'https://www.googleapis.com/auth/userinfo.email',
   'https://www.googleapis.com/auth/cloud-platform',
 ];
 
+// Path to the root of the gcloud configuration.
+function gcloudDir(): string {
+  return path.join(appSettings.datalabRoot, '/content/datalab/.config');
+}
+
+// Path for user-specific credentials.
+function userCredFile(): string {
```

Comment:
Ah. Okay. Thanks for explaining this. Maybe someday we will have another run_no_docker.sh.